aicsimageio/readers/array_like_reader.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from typing import Optional, Tuple
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+
+from .. import constants, exceptions
+from ..dimensions import DimensionNames
+from ..metadata import utils as metadata_utils
+from ..types import ArrayLike
+from .reader import Reader
+
+###############################################################################
+
+
+class ArrayLikeReader(Reader):
+    """
+    A catch all for numpy or dask to Reader interface.
+
+    Useful for passing through raw data to Reader -> Writer.
+
+    Parameters
+    ----------
+    image: types.ArrayLike
+        A numpy ndarray or dask Array.
+
+    known_dims: Optional[str]
+        A string containing dimension identifiers to use instead of guesses.
+        Default: None (guess dimensions)
+    """
+
+    @staticmethod
+    def _is_supported_image(image: ArrayLike, *args, **kwargs) -> bool:
+        return isinstance(image, (np.ndarray, da.Array))
+
+    def __init__(self, image: ArrayLike, known_dims: Optional[str] = None, **kwargs):
+        # Enforce valid image
+        if not self._is_supported_image(image):
+            raise exceptions.UnsupportedFileFormatError(self.__class__.__name__, image)
+
+        # Store dims
+        if known_dims is None:
+            dims = self._guess_dim_order(image.shape)
+        else:
+            dims = known_dims
+
+        # Turn dims to list
+        dims = [d for d in dims]
```

Comment:
Is there a reason for this approach over?
```suggestion
        dims = list(dims)
```

Reply:
Because I didn't know about this! / forgot that this is possible.